runtime/src/iree/base/internal/cpu.c

```diff
@@ -310,6 +310,74 @@ static void iree_cpu_initialize_from_platform_x86_64(uint64_t* out_fields) {
   out_fields[0] = out0;
 }
 
+#elif defined(IREE_ARCH_RISCV_64)
+#if defined(IREE_PLATFORM_ANDROID) || defined(IREE_PLATFORM_LINUX)
+
+// First 4 characters are "rv32" or "rv64"
+#define IREE_RISCV_64_ISA_SEARCH_START 4
+
+static void iree_cpu_initialize_from_platform_riscv_64(uint64_t* out_fields) {
+  char* line;
+  char isa[1024];
+  char* cpuinfo_path = "/proc/cpuinfo";
```

Comment:
Maybe there is no need for a `cpuinfo_path` variable, just hardcode the path in the fopen call.  If you did add a variable for it, please make it `const char[]`.

Reply:
Note @benvanik :  some googling suggests that there indeed isn't a better alternative than parsing `/proc/cpuinfo`.  This does the same thing:
https://github.com/google/cpu_features/blob/main/src/impl_riscv_linux.c

Reply:
I found we can use hwprobe to check vector extension. The vector information is added in linux v6.5. It is really new...
Do we prefer to use hwprobe or keep using this parsing `/proc/cpuinfo` method?
https://www.kernel.org/doc/html/v6.5/riscv/hwprobe.html

Reply:
Either way could be problematic running Linux user space qemu (IREE CI setup). I am not sure you can either call `hwcap` or check `/proc/cpuinfo` within QEMU.

For RISCV, shouldn't you be able to read `mstatus` and other CSRs to check if the vector extension is enabled with what configuration, such as vlenb?

Reply:
User mode qemu has a fake `/proc/cpuinfo`. `hwcap` seems support too. But maybe it need newer qemu version.
https://github.com/qemu/qemu/commit/8ddc171b7b302844d9f4598125fed925b72c686c

Read vlenb without v extension will cause illegal instruciton. The `mstatus` can not access in userspace. So seems it can not get this information from reading CSRs.

Reply:
> Do we prefer to use hwcap or keep using this parsing /proc/cpuinfo method?

If hwcap is possible, that is my preference. Far easier to check for correctness and fewer possible failure points than file IO and parsing.

Otherwise, whatever works for you.

On Arm, Qemu has supported hwcap very well.

I think mostly it depends if the requirement of using a sufficiently new Linux kernel and Qemu version is OK with your needs.

Reply:
I agree using hwcap is much  more safer and efficient than parsing `/proc/cpuinfo`. Let me try it on the latest qemu. If it works I'll send another PR and drop this one. If qemu does not support hwcap I'll fix this PR.